packages/plugin-eslint/src/lib/runner/index.ts

```diff
@@ -47,22 +47,23 @@ export async function executeRunner(argv = process.argv): Promise<void> {
   await writeFile(RUNNER_OUTPUT_PATH, JSON.stringify(audits));
 }
 
-export function createRunnerConfig(
+export async function createRunnerConfig(
   scriptPath: string,
   audits: Audit[],
   eslintrc: string,
   patterns: string | string[],
-): RunnerConfig {
+): Promise<RunnerConfig> {
+  const config: ESLintPluginRunnerConfig = {
+    eslintrc,
+    slugs: audits.map(audit => audit.slug),
+    patterns: toArray(patterns),
+  };
+  await ensureDirectoryExists(WORKDIR);
```

Comment:
This way it'll infer what directory needs to be created from `PLUGIN_CONFIG_PATH` (used like this in `executeRunner`):

```suggestion
  await mkdir(dirname(PLUGIN_CONFIG_PATH), { recursive: true });
```

Reply:
Why not keep `ensureDirectoryExists`? It does the same thing plus it does not throw when the directory is already created.

Reply:
> It does the same thing plus it does not throw when the directory is already created.

But neither does `mkdir('...', { recursive: true })`, AFAIK.

> Why not keep `ensureDirectoryExists` for the new path?

Although I don't actually see the point of the `ensureDirectoryExists` wrapper, the main suggestion is to use `dirname(PLUGIN_CONFIG_PATH)` instead of `WORKDIR`. That way it's more directly linked to the file you want to create there. And if we decide to not put the `plugin-config.json` directly under `WORKDIR`, we can change without touching this logic. 

You could even extract those two statements into a helper function:
```ts
await mkdir(dirname(filePath), { recursive: true });
await writeFile(filePath, content);
```

Reply:
I see. Thought using helpers consistently is the goal. Perhaps we don't need the helper then. But that is a story for another time.